core/management/commands/initialize_historical_counters.py

```diff
@@ -0,0 +1,125 @@
+from funcy import zipdict, isums, walk_values, group_by, count_by, second, compose, first
+from tqdm import tqdm
+from handy.db import queryset_iterator
+from datetime import datetime
+from django.db.models import Count
+from django.db import transaction
+from django.core.management import BaseCommand
+from core.models import HistoricalCounter
+from legacy.models import Series, Sample, Platform, PlatformProbe
+from core.models import User
+from tags.models import Tag, SerieAnnotation, SampleAnnotation
+
+MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+
+def floor_attrs_date(model):
+    month, day, year = model.attrs.get('submission_date', 'Jan 1 1960').split()
+    if month == 'Jans':
+        month = 'Jan'
+    return datetime(int(year), MONTHS.index(month) + 1, 1)
+
+def floor_date(date):
+    return datetime(date.year, date.month, 1)
+
+
+START_DATE = datetime(2014, 10, 1)
+CURRENT_DATE = floor_date(datetime.now())
+
+def accumulate(data):
+    dates, counts = zip(*sorted(data.items()))
+    return zipdict(dates, isums(counts))
+
+def gather_by(fn, iterator, date_floor):
+    def _gather_fn(array):
+        return sum(map(fn, array))
+    return walk_values(_gather_fn, group_by(date_floor, iterator))
+
+def get_value(keys, index):
+    """
+    There is no secuence with two or more holes in data array,
+    so we can do only one step back
+    """
+    def _getter(item):
+        return item.get(
+            keys[index],
+            item.get(keys[index - 1 if index > 0 else index], 0))
+    return _getter
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        series = accumulate(count_by(floor_attrs_date, Series.objects.all()))
+
+        iterator = tqdm(queryset_iterator(Sample.objects.all(), 30000),
+                        total=Sample.objects.count(),
+                        desc='samples')
+        samples = accumulate(count_by(floor_attrs_date, iterator))
+
+        iterator = tqdm(Platform.objects.all().annotate(probes_count=Count('probes')).iterator(),
+                        total=Platform.objects.count(),
+                        desc='platforms')
+        platforms_data = [
+            {
+                'created_on': min(map(floor_attrs_date, platform.sample_set.all())),
```

Comment:
You are like a baby )
Obvious optimizations:
- user `.values_list().iterator()` instead of `.all()`.
- select sample with min id then get its date
- fetch all samples at once, not for each platform (or select only those with min id in platform)

Less obvious:
- do not fetch samples, use series instead
- fetch all of them and map platform -> serie, use it to look up locally